### ISSUE TITLE
Bug querying building footprints

### ISSUE THREAD

**Issue by Eph97** [gboeing/osmnx#543]
**Problem description**
I have been using osmnx for a couple of months now as part of a research project. I recently updated to the new version of osmnx and keep getting an error when I try to query building footprints. It was however working before I updated. I continue to get the error message "pygeos.GEOSException: IllegalArgumentException: Argument must be Polygonal or LinearRing" when I am fairly confident I am passing a valid Multipolygon (I've also tried with normal polygon's and get the same error).I have also tried querying building footprints using from_place and from_point and neither have worked for me either. Querying street networks and generating graphs using this polygon have all worked. It's only when I try to retrieve the building footprint that osmnx fails and this only started failing after I updated. 

I've attached a zip file containing example data.

**Environment information**
 I am using a mac running Catalina (10.15.2)
I'm using python 3.8.5 installed using conda
and what I believe is the latest osmnx version (0.15.1)

<details>

# packages in environment at /Users/ephraimsutherland/opt/miniconda3/envs/ox:
#
# Name                    Version                   Build  Channel
affine                    2.3.0                    pypi_0    pypi
appnope                   0.1.0           py38h32f6830_1001    conda-forge
astroid                   2.4.2                    pypi_0    pypi
attrs                     19.3.0                     py_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.1                      py_0    conda-forge
bleach                    3.1.5              pyh9f0ad1d_0    conda-forge
boost-cpp                 1.72.0               hef959ae_2    conda-forge
branca                    0.4.1                      py_0    conda-forge
brotlipy                  0.7.0           py38h64e0658_1000    conda-forge
bzip2                     1.0.8                h0b31af3_2    conda-forge
ca-certificates           2020.6.20            hecda079_0    conda-forge
cairo                     1.16.0            ha8983da_1005    conda-forge
cartopy                   0.18.0                   pypi_0    pypi
certifi                   2020.6.20        py38h32f6830_0    conda-forge
cffi                      1.14.1           py38hc4dd44e_0    conda-forge
cfitsio                   3.470                hdf94aef_6    conda-forge
chardet                   3.0.4           py38h32f6830_1006    conda-forge
click                     7.1.2              pyh9f0ad1d_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.5.0                      py_0    conda-forge
contextily                1.0.0                    pypi_0    pypi
cryptography              3.0              py38h52adbb4_0    conda-forge
curl                      7.71.1               hbdca1c0_1    conda-forge
cycler                    0.10.0                     py_2    conda-forge
cykhash                   1.0.2                    pypi_0    pypi
cython                    0.29.20                  pypi_0    pypi
decorator                 4.4.2                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
descartes                 1.1.0                      py_4    conda-forge
entrypoints               0.3             py38h32f6830_1001    conda-forge
expat                     2.2.9                h4a8c4bd_2    conda-forge
fiona                     1.8.13           py38hcfd527e_2    conda-forge
folium                    0.11.0                     py_0    conda-forge
fontconfig                2.13.1            h79c0d67_1002    conda-forge
freetype                  2.10.2               h8da9a1a_0    conda-forge
freexl                    1.0.5             h0b31af3_1002    conda-forge
gdal                      3.1.2            py38h7882b60_0    conda-forge
geopandas                 0.8.1                      py_0    conda-forge
geoplot                   0.4.1                    pypi_0    pypi
geos                      3.8.1                h4a8c4bd_0    conda-forge
geotiff                   1.6.0                hd8796ba_0    conda-forge
gettext                   0.19.8.1          h46ab8bc_1002    conda-forge
giflib                    5.2.1                h0b31af3_2    conda-forge
glib                      2.65.0               h577aef8_0    conda-forge
hdf4                      4.2.13            h84186c3_1003    conda-forge
hdf5                      1.10.6          nompi_h3e39495_100    conda-forge
icu                       67.1                 h4a8c4bd_0    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
importlib-metadata        1.7.0            py38h32f6830_0    conda-forge
importlib_metadata        1.7.0                         0    conda-forge
ipykernel                 5.3.4            py38h23f93f0_0    conda-forge
ipython                   7.16.1           py38h23f93f0_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
isort                     4.3.21                   pypi_0    pypi
jedi                      0.17.2           py38h32f6830_0    conda-forge
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
joblib                    0.16.0                     py_0    conda-forge
jpeg                      9d                   h0b31af3_0    conda-forge
json-c                    0.13.1            h575e443_1002    conda-forge
json5                     0.9.4              pyh9f0ad1d_0    conda-forge
jsonschema                3.2.0            py38h32f6830_1    conda-forge
jupyter_client            6.1.6                      py_0    conda-forge
jupyter_core              4.6.3            py38h32f6830_1    conda-forge
jupyterlab                2.1.4                      py_0    conda-forge
jupyterlab_server         1.2.0                      py_0    conda-forge
kealib                    1.4.13               h40102fb_1    conda-forge
kiwisolver                1.2.0            py38ha0d09dd_0    conda-forge
krb5                      1.17.1               h14dd6a4_1    conda-forge
lazy-object-proxy         1.4.3                    pypi_0    pypi
lcms2                     2.11                 h174193d_0    conda-forge
libblas                   3.8.0               17_openblas    conda-forge
libcblas                  3.8.0               17_openblas    conda-forge
libcurl                   7.71.1               he6690cf_1    conda-forge
libcxx                    10.0.1               h5f48129_0    conda-forge
libdap4                   3.20.6               h993cace_1    conda-forge
libedit                   3.1.20191231         hed1e85f_1    conda-forge
libffi                    3.2.1             h4a8c4bd_1007    conda-forge
libgdal                   3.1.2                h6ef84d7_0    conda-forge
libgfortran               4.0.0                         2    conda-forge
libiconv                  1.15              h0b31af3_1006    conda-forge
libkml                    1.3.0             h169b8f9_1011    conda-forge
liblapack                 3.8.0               17_openblas    conda-forge
libnetcdf                 4.7.4           nompi_hc5b2cf3_105    conda-forge
libopenblas               0.3.10          openmp_h63d9170_4    conda-forge
libpng                    1.6.37               hbbe82c9_1    conda-forge
libpq                     12.3                 h489d428_0    conda-forge
libsodium                 1.0.17               h01d97ff_0    conda-forge
libspatialindex           1.9.3                h4a8c4bd_3    conda-forge
libspatialite             4.3.0a            h658e6c1_1038    conda-forge
libssh2                   1.9.0                h39bdce6_4    conda-forge
libtiff                   4.1.0                h2ae36a8_6    conda-forge
libwebp-base              1.1.0                h0b31af3_3    conda-forge
libxml2                   2.9.10               h7fdee97_2    conda-forge
llvm-openmp               10.0.1               h28b9765_0    conda-forge
lz4-c                     1.9.2                h4a8c4bd_1    conda-forge
mapclassify               2.3.0                    pypi_0    pypi
markupsafe                1.1.1            py38h64e0658_1    conda-forge
matplotlib-base           3.3.0            py38haed9462_1    conda-forge
mccabe                    0.6.1                    pypi_0    pypi
mercantile                1.1.5                    pypi_0    pypi
mistune                   0.8.4           py38h64e0658_1001    conda-forge
munch                     2.5.0                      py_0    conda-forge
nbconvert                 5.6.1            py38h32f6830_1    conda-forge
nbformat                  5.0.7                      py_0    conda-forge
ncurses                   6.2                  hb1e8313_1    conda-forge
networkx                  2.4                        py_1    conda-forge
notebook                  6.0.3            py38h32f6830_1    conda-forge
numpy                     1.19.1           py38h598c1e0_0    conda-forge
olefile                   0.46                       py_0    conda-forge
openjpeg                  2.3.1                h254dc36_3    conda-forge
openssl                   1.1.1g               h0b31af3_0    conda-forge
osmnx                     0.15.1             pyh9f0ad1d_0    conda-forge
packaging                 20.4               pyh9f0ad1d_0    conda-forge
pandas                    1.0.5            py38h5fc983b_0    conda-forge
pandoc                    2.10.1               haf1e3a3_0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.7.1              pyh9f0ad1d_0    conda-forge
pcre                      8.44                 h4a8c4bd_0    conda-forge
pexpect                   4.8.0            py38h32f6830_1    conda-forge
pickleshare               0.7.5           py38h32f6830_1001    conda-forge
pillow                    7.2.0            py38h83dc5e5_1    conda-forge
pip                       20.1.1                     py_1    conda-forge
pixman                    0.38.0            h01d97ff_1003    conda-forge
poppler                   0.89.0               h3232a60_1    conda-forge
poppler-data              0.4.9                         1    conda-forge
postgresql                12.3                 h62ab893_0    conda-forge
proj                      7.0.0                h45baca5_5    conda-forge
prometheus_client         0.8.0              pyh9f0ad1d_0    conda-forge
prompt-toolkit            3.0.5                      py_1    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pygeos                    0.7.1                    pypi_0    pypi
pygments                  2.6.1                      py_0    conda-forge
pylint                    2.5.3                    pypi_0    pypi
pyopenssl                 19.1.0                     py_1    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyproj                    2.6.1.post1      py38h7d7e87d_0    conda-forge
pyrobuf                   0.9.3                    pypi_0    pypi
pyrosm                    0.5.2                    pypi_0    pypi
pyrsistent                0.16.0           py38h64e0658_0    conda-forge
pyshp                     2.1.0                    pypi_0    pypi
pysocks                   1.7.1            py38h32f6830_1    conda-forge
python                    3.8.5           h85f3143_1_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python-rapidjson          0.9.1                    pypi_0    pypi
python_abi                3.8                      1_cp38    conda-forge
pytz                      2020.1             pyh9f0ad1d_0    conda-forge
pyzmq                     19.0.1           py38h1fcdcd6_0    conda-forge
rasterio                  1.1.5                    pypi_0    pypi
readline                  8.0                  h0678c8f_2    conda-forge
requests                  2.24.0             pyh9f0ad1d_0    conda-forge
rtree                     0.9.4            py38h08f867b_1    conda-forge
scikit-learn              0.23.1           py38hb7f0800_0    conda-forge
scipy                     1.5.2            py38h1402333_0    conda-forge
seaborn                   0.10.1                   pypi_0    pypi
send2trash                1.5.0                      py_0    conda-forge
setuptools                49.2.0           py38h32f6830_0    conda-forge
shapely                   1.7.0            py38h5b1fbde_3    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
snuggs                    1.4.7                    pypi_0    pypi
sqlite                    3.32.3               h93121df_1    conda-forge
tbb                       2019.9               ha1b3eb9_1    conda-forge
terminado                 0.8.3            py38h32f6830_1    conda-forge
testpath                  0.4.4                      py_0    conda-forge
threadpoolctl             2.1.0              pyh5ca1d4c_0    conda-forge
tiledb                    2.0.6                h84aa2a7_0    conda-forge
tk                        8.6.10               hbbe82c9_0    conda-forge
toml                      0.10.1                   pypi_0    pypi
tornado                   6.0.4            py38h64e0658_1    conda-forge
traitlets                 4.3.3            py38h32f6830_1    conda-forge
tzcode                    2020a                h0b31af3_0    conda-forge
urllib3                   1.25.10                    py_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.34.2                     py_1    conda-forge
wrapt                     1.12.1                   pypi_0    pypi
xerces-c                  3.2.3                h0a0444a_1    conda-forge
xz                        5.2.5                h0b31af3_1    conda-forge
zeromq                    4.3.2                h6de7cb9_2    conda-forge
zipp                      3.1.0                      py_0    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge
zstd                      1.4.5                h0384e3a_2    conda-forge

</details>

[Isochrone.zip](https://github.com/gboeing/osmnx/files/5003045/Isochrone.zip)

<img width="1440" alt="Screen Shot 2020-07-30 at 1 48 17 PM" src="https://user-images.githubusercontent.com/22008945/88956625-a03e6a00-d26b-11ea-8f7f-4a4ff0b22774.png">
```python



```

import geopandas
import osmnx as ox

All_Isochrones = geopandas.read_file('/Users/<user>/documents/Isochrone/Isochrone.shp')

footprint = ox.footprints.footprints_from_polygon(All_Isochrones.loc[[0], 'geometry'].values[0], footprint_type="building", retain_invalid=False)

Traceback (most recent call last):
  File "bug.py", line 6, in <module>
    footprint = ox.footprints.footprints_from_polygon(All_Isochrones.loc[[0], 'geometry'].values[0], footprint_type="building", retain_invalid=False)
  File "/Users/ephraimsutherland/opt/miniconda3/envs/ox/lib/python3.8/site-packages/osmnx/footprints.py", line 501, in footprints_from_polygon
    return _create_footprints_gdf(polygon, footprint_type, retain_invalid)
  File "/Users/ephraimsutherland/opt/miniconda3/envs/ox/lib/python3.8/site-packages/osmnx/footprints.py", line 126, in _create_footprints_gdf
    filter1 = gdf["geometry"].is_valid
  File "/Users/ephraimsutherland/opt/miniconda3/envs/ox/lib/python3.8/site-packages/geopandas/base.py", line 170, in is_valid
    return _delegate_property("is_valid", self)
  File "/Users/ephraimsutherland/opt/miniconda3/envs/ox/lib/python3.8/site-packages/geopandas/base.py", line 74, in _delegate_property
    data = getattr(a_this, op)
  File "/Users/ephraimsutherland/opt/miniconda3/envs/ox/lib/python3.8/site-packages/geopandas/array.py", line 400, in is_valid
    return vectorized.is_valid(self.data)
  File "/Users/ephraimsutherland/opt/miniconda3/envs/ox/lib/python3.8/site-packages/geopandas/_vectorized.py", line 457, in is_valid
    return pygeos.is_valid(data)
  File "/Users/ephraimsutherland/opt/miniconda3/envs/ox/lib/python3.8/site-packages/pygeos/predicates.py", line 258, in is_valid
    result = lib.is_valid(geometry, **kwargs)
pygeos.GEOSException: IllegalArgumentException: Argument must be Polygonal or LinearRing

**Comment by gboeing** [comment]
@Eph97 your code snippet works for me:

```python
import geopandas
import osmnx as ox
ox.config(use_cache=True, log_console=True)

isochrones = geopandas.read_file('./Isochrone/Isochrone.shp')
geom = isochrones.loc[[0], 'geometry'].values[0]
fp = ox.footprints_from_polygon(geom, footprint_type="building", retain_invalid=False)
fp.shape #(285945, 547)
```

This produces a GeoDataFrame `fp` with 285,945 rows and 547 columns. I have tested this with both v0.15.1 and with the current master branch.

I wonder if there's something strange in your installation... does everything else with OSMnx work ok on your system? Would you mind trying your code again using the OSMnx [docker image](https://hub.docker.com/r/gboeing/osmnx) and seeing if it works ok for you there?

**Comment by Eph97** [comment]
@gboeing Thank you very much for your help. It seems it may have been an issue with the dependencies when I updated osmnx using conda. After seeing that it worked on your installation, I deleted my environment and reinstalled everything using conda and it seems to work now. I didn't initially think to do so because everything else seemed to be working properly. Hopefully this helps someone else.

Also, as everyone else seems to say, I love osmnx. It's made a lot of my work much easier over the past couple of months
